cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/Storage.java

```diff
@@ -70,4 +71,13 @@ public interface Storage {
    */
   String allocateTableLocation(
       String databaseId, String tableId, String tableUUID, String tableCreator);
+
+  /**
+   * Deallocates/deletes Table Storage location
+   *
+   * @param location the base location of the table
+   * @param tableCreator the creator of the table
```

Comment:
This need not tableCreator necessarily for all underlying storages.  May be actingPrincipal?